pyccel/codegen/printing/pycode.py

```diff
@@ -175,6 +175,14 @@ def _print_NativeComplex(self, expr):
     def _print_Variable(self, expr):
         return self._print(expr.name)
 
+    def _print_DottedVariable(self, expr):
+        rhs_code = self._print_Variable(expr)
+        lhs_code = self._print(expr.lhs)
+        return f"{lhs_code}.{rhs_code}"
+
+    def _print_DottedFunctionCall(self, expr):
+        return (f"{str(expr.func_name)}({', '.join([str(arg) for arg in expr.args])})\n")
```

Comment:
`join` can take an iterable argument which avoids unnecessary memory allocation

```suggestion
        return (f"{str(expr.func_name)}({', '.join(str(arg) for arg in expr.args)})\n")
```

Reply:
You should use `self._print` to print objects, not `str()`